### PR TITLE
Fixed reloading data when change managing entity is sucessfully completes

### PR DESCRIPTION
**Pull request by aurangzaibumer**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes - https://github.com/opensrp/fhir-resources/issues/1742

Testing reference

https://github.com/opensrp/fhircore/assets/35099184/3664e3f9-4898-42ba-a177-b32e55bde946

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesListener.kt [hunk 1]
```diff
@@ -65,7 +65,7 @@ abstract class RulesListener : RuleListener {
             exception,
             "${exception.localizedMessage}, consider checking for null before usage: e.g ${exception.variable} != null"
           )
-        else -> log(exception)
+        else -> log(exception, "failed to execute rule : ${rule.name} ")
```
Comment (aurangzaibumer) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesListener.kt:68]: added just to print the rule name when fails

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModel.kt [hunk 2]
```diff
@@ -80,6 +81,8 @@ constructor(
   private val listResourceDataStateMap =
     mutableStateMapOf<String, SnapshotStateList<ResourceData>>()
 
+  val shouldReloadData = MutableLiveData<Boolean>()
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModel.kt:84]: Why do we need a new LiveData just to refresh the profile when dialog is dismissed?

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt [hunk 3]
```diff
@@ -67,6 +67,14 @@ class ProfileFragment : Fragment() {
         }
       }
     }
+    // this will be called when change managing entity is completed to reload data
+    profileViewModel.shouldReloadData.observe(viewLifecycleOwner) {
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt:71]: Do we need a new live data?

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt [hunk 4]
```diff
@@ -101,6 +110,12 @@ class ProfileFragment : Fragment() {
     }
   }
 
+  private suspend fun handleRefreshLiveData() {
+    with(profileFragmentArgs) {
+      profileViewModel.retrieveProfileUiState(profileId, resourceId, resourceConfig, params)
+    }
+  }
+
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt:118]: I think we can move the function block (and delete this private function) to where this is called as this private function is only called from one place.

**Review by ellykits — CHANGES_REQUESTED** [review]

**Review by ellykits — DISMISSED** [review]

**Review by pld — APPROVED** [review]
